Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,450 @@
+            }
+        }
+    }
+
+    Context 'When getting a codespace for a specified codespace name' {
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace
+            $codespace = Get-GitHubCodespace -CodespaceName $codespaces[0].name
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespace.PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return the correct properties' {
+            $codespace.owner.login | Should -Be $script:OwnerName
+        }
+    }
+
+    Context 'When specifiying the Uri parameter' {
+        BeforeAll {
+            $codespace = Get-GitHubCodespace -Uri $repo.RepositoryUrl
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespace.PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return the correct properties' {
+            $codespace.owner.login | Should -Be $script:OwnerName
+            $codespace.repository.name | Should -Be $repo.name
+        }
+    }
+
+    Context "When specifiying the Uri parameter from the pipeline" {
+        BeforeAll {
+            $codespace = $repo | Get-GitHubCodespace
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespace.PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return the correct properties' {
+            $codespace.repository.name | Should -Be $repo.name
+        }
+    }
+}
+
+
+Describe 'GitHubCodespaces\New-GitHubCodespace' {
+
+    Context -Name 'When creating a repository for the authenticated user' {
+
+        Context -Name 'When creating a codespace with default settings with RepositoryId' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    RepositoryId = $repo.Id
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+                Start-Sleep -Seconds 2
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.repository.name | Should -Be $repo.name
+                $codespace.idle_timeout_minutes | Should -Be 30
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with default settings with Ref' {
+            BeforeAll {
+                $repoWithPR = Get-GitHubRepository -OrganizationName 'super-fake-not-real-org' |
```

Comment:
Wait -- why would this test work at all given that this org doesn't exist?

Reply:
It does exist as a sandbox in my GitHub Enterprise environment.  But, this should use $script:organizationName based on test config Settings.  I'll correct it.